prow/github/report/report.go

```diff
@@ -189,6 +201,17 @@ func Report(ghc GithubClient, reportTemplate *template.Template, pj prowapi.Prow
 	return nil
 }
 
+// findSHAInPRCommits returns true if a given SHA in found in a slice of RepositoryCommits.
+func findSHAInPRCommits(sha string, icomm []github.RepositoryCommit) bool {
+	for _, c := range icomm {
+		if c.SHA != sha {
+			continue
+		}
+		return true
+	}
+	return false
+}
+
 // parseIssueComments returns a list of comments to delete, a list of table
```

Comment:
I would be nice to break apart this function a bit into pieces we can unit test better. these for loops are hard to understand